### DISCUSSION TITLE
启动dble之后，日志会报" found MySQL master/slave Replication err "

### DISCUSSION THREAD

**Discussion by nejinn** [actiontech/dble#3104] [Q&A] [answered]
* **dble version：3.21.10.0-20211119065525**  

**db.xml**  

```
<dble:db xmlns:dble="http://dble.cloud/" version="4.0">
    <dbGroup name="dbGroup1" rwSplitMode="3" delayThreshold="100">
        <heartbeat errorRetryCount="1" timeout="2000">show slave status</heartbeat>
        <dbInstance name="instanceM1" url="ip:3306" user="root" password="******" maxCon="1000" minCon="10"
                    primary="true" readWeight="1" id="79">
            <property name="testOnCreate">true</property>
        </dbInstance>
        <dbInstance name="instanceM2" url="ip:3306" user="root" password="******" maxCon="1000" minCon="10"
                    primary="false" readWeight="1" id="85">
            <property name="testOnCreate">true</property>
        </dbInstance>
        <dbInstance name="instanceM3" url="10.75.142.86:3306" user="root" password="******" maxCon="1000" minCon="10"
                    primary="false" readWeight="1" id="86">
            <property name="testOnCreate">true</property>
        </dbInstance>
    </dbGroup>
</dble:db>


```

日志中频繁出现：

> WARN [complexQueryExecutor5] (com.actiontech.dble.backend.heartbeat.MySQLDetector.setStatusBySlave(MySQLDetector.java:198)) - found MySQL master/slave Replication err !!! DbInstanceConfig [hostName=instanceM3, url=10.75.142.86:3306]

**Comment by PanternBao** [comment, marked as answer]
## 前置知识
参考：https://github.com/actiontech/dble-docs-cn/blob/master/1.config_file/1.04_db.xml.md dbGroup中的heartbeat 属性。

设置心跳语句后，dble会使用该语句定期对MySQL实例进行探活。

心跳语句有以下几种建议值:
1. 普通心跳只是用于探活，建议使用select 1
2. 使用 select @@read_only 探测结点可用性以及可写性
3. 使用show slave status

###  show slave status 心跳的功用
1. 通过判断是否返回结果判断实例是否正常
2. 再通过返回的结果检查复制是否正常
3. 延迟检测。如果Seconds_Behind_Master返回的状态有延迟，那么会被记入mysql实例的主从延迟中，影响读请求的路由分发，延迟超过指定限制读写分离会变为只从主库读取数据。 Seconds_Behind_Master 结果可能不准确，需要注意。

## 结论
日志中的报错，只有当使用 show slave status 作为心跳语句才会出现，此日志表示，配置的mysql组复制关系异常，需要运维介入修复复制关系。

**Reply by Dinosauria** [reply]
需要注意一下，当使用的mysql后端的架构为MGR（组复制），且心跳sql使用为 show slave status;时，在dble启动后，dble.log也会出现如上的报错，原因是 组复制中查询复制（online）状态正常与否，是不能通过使用sql: show slave status;来获得的。

**Comment by wangjuan-action** [comment]
如果遇到MySQL主从延时值过大或反复跳动，可参考 https://mp.weixin.qq.com/s/E1Pg3zq7Zb70oRbsGGaTjg 进行排查
